### PR TITLE
added new e2e tests to cover new functionality

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

File: __tests__/selenium_tests/e2e_tests.ts [hunk 1]
```diff
@@ -187,7 +351,7 @@ describe('End to end test', () => {
 
       logger.info('sending deployment information')
       const deploymentData = await fs.readFile(deploymentMetadataPath, { encoding: 'utf-8' })
-      const deploymentInfo = JSON.parse(deploymentData)
+      const deploymentInfo = { ...JSON.parse(deploymentData) }
```
Comment (a3957273) [inline __tests__/selenium_tests/e2e_tests.ts:354]: What's the purpose of this expansion?

File: __tests__/selenium_tests/e2e_tests.ts [hunk 2]
```diff
@@ -198,6 +362,9 @@ describe('End to end test', () => {
 
       logger.info(`found url contains deployment`)
       await driver.wait(until.urlContains('/deployment/'))
+      await driver.sleep(2000)
```
Comment (a3957273) [inline __tests__/selenium_tests/e2e_tests.ts:365]: We already wait until the deployment contains `/deployment/`, why are we waiting more?

Comment (ARADDCC002) [reply __tests__/selenium_tests/e2e_tests.ts:365]: This was causing issues, but I can take it out and try again. Sometimes the tests don't wait until the components fully render, and I'm not sure how else to do it.

File: __tests__/selenium_tests/e2e_tests.ts (hunk 1 above)
Comment (ARADDCC002) [reply __tests__/selenium_tests/e2e_tests.ts:354]: I think I changed this to something else then changed it back! Will revert.

File: __tests__/selenium_tests/e2e_tests.ts (hunk 2 above)
Comment (ARADDCC002) [reply __tests__/selenium_tests/e2e_tests.ts]: Seems to be working without it :)

**Review by a3957273 — APPROVED** [review]

**Pull request merged**